services/introspection/local.go

```diff
@@ -94,9 +91,16 @@ func (l *Local) Plugins(ctx context.Context, req *api.PluginsRequest, _ ...grpc.
 }
 
 func (l *Local) getPlugins() []api.Plugin {
+	if !plugin.AreRegisteredPluginsInitialized(l.plugins) {
```

Comment:
Instead of checking against registered, can you just check the size against what is already cached. If it changes, then update the cache.